packages/eslint-plugin/lib/rules/require-img-alt.js

```diff
@@ -77,16 +142,37 @@ module.exports = {
 /**
  * @param {Tag} node
  * @param {string[]} substitute
- * @returns
+ * @returns {{hasAnyAlt: boolean, hasEmptyAlt: boolean, hasValidContent: boolean}}
  */
-function hasAltAttrAndValue(node, substitute = []) {
-  return node.attributes.some((attr) => {
-    if (attr.key && attr.value) {
-      return (
-        (attr.key.value === "alt" || substitute.includes(attr.key.value)) &&
-        typeof attr.value.value === "string"
-      );
+function hasValidAltOrSubstitute(node, substitute = []) {
+  let hasAnyAlt = false;
+  let hasEmptyAlt = false;
+  let hasValidContent = false;
+
+  for (const attr of node.attributes) {
+    if (attr.key && attr.key.value) {
+      const isAltAttr = attr.key.value === "alt";
+      const isSubstituteAttr = substitute.includes(attr.key.value);
+
+      if (isAltAttr || isSubstituteAttr) {
+        hasAnyAlt = true;
+
+        if (
+          attr.value &&
+          typeof attr.value.value === "string" &&
+          attr.value.value.trim() !== ""
+        ) {
+          hasValidContent = true;
```

Comment:
As mentioned earlier, the empty string ("") is also a valid value. It looks like we need a refactoring.